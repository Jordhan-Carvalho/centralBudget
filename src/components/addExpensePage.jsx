import React, { Component } from "react";
import { connect } from "react-redux";
import ExpenseForm from "./expenseForm";
import { addExpense } from "../actions/expenses";

export class AddExpensePage extends Component {
  onSubmit = expense => {
    this.props.addExpense(expense);
    this.props.history.push("/");
  };

  render() {
    return (
      <React.Fragment>
        <h1>Add exprense</h1>
        <ExpenseForm onSubmit={this.onSubmit} />
      </React.Fragment>
    );
  }
}

// to simplify props.dispatch(addExpense(expense)) to props.onSubmit(expense)
const mapDispatchToProps = dispatch => {
  return {
    addExpense: expense => dispatch(addExpense(expense))
  };
};

export default connect(
  undefined,
  mapDispatchToProps
)(AddExpensePage);
