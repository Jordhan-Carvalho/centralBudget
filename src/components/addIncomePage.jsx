import React, { Component } from "react";
import { connect } from "react-redux";
import Form from "./common/form";
import { startAddIncome } from "../actions/incomes";

export class AddIncomePage extends Component {
  onSubmit = income => {
    this.props.startAddIncome(income);
    this.props.history.push("/dashboard");
  };

  render() {
    return (
      <React.Fragment>
        <h1>Add Income</h1>
        <Form onSubmit={this.onSubmit} />
      </React.Fragment>
    );
  }
}

// to simplify props.dispatch(addExpense(expense)) to props.onSubmit(expense)
const mapDispatchToProps = dispatch => {
  return {
    startAddIncome: income => dispatch(startAddIncome(income))
  };
};

export default connect(
  undefined,
  mapDispatchToProps
)(AddIncomePage);
