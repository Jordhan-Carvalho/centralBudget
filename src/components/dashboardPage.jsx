import React, { Component } from "react";
import TransChart from "./transChart";
import SavingsChart from "./savingsChart";
import ListFilters from "./common/listFilters";
import TransSumary from "./common/transSumary";

class DashboardPage extends Component {
  render() {
    return (
      <React.Fragment>
        <div className="row justify-content-center">
          <div className="col-12 boxdash">
            <TransChart height={200} />
          </div>
        </div>

        <div className="row justify-content-center">
          <div className="col-12 boxdash">
            <SavingsChart height={200} />
          </div>
        </div>

        <div className="row">
          <div className="col-12 boxdash">
            <ListFilters />
            <TransSumary type={"expense"} />
            <TransSumary type={"income"} />
            <TransSumary type={"saving"} />
          </div>
        </div>
      </React.Fragment>
    );
  }
}

export default DashboardPage;
