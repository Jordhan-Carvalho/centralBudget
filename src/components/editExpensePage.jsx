import React, { Component } from "react";
import { connect } from "react-redux";
import Form from "./common/form";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";

export class EditExpensePage extends Component {
  onRemove = () => {
    this.props.startRemoveExpense(this.props.expense);
    this.props.history.push("/");
  };

  onSubmit = expense => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };

  render() {
    return (
      <div className="form-group">
        <h1>Edit {this.props.expense.description}</h1>
        <div className="form-group">
          <Form expense={this.props.expense} onSubmit={this.onSubmit} />
        </div>
        <div className="form-group">
          <button className="btn btn-danger" onClick={this.onRemove}>
            Remove
          </button>
        </div>
      </div>
    );
  }
}

// maps

const mapDispatchToProps = dispatch => {
  return {
    startEditExpense: (expenseID, expense) =>
      dispatch(startEditExpense(expenseID, expense)),
    startRemoveExpense: expense => dispatch(startRemoveExpense(expense))
  };
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(expense => {
      return expense.id === props.match.params.id;
    })
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpensePage);
