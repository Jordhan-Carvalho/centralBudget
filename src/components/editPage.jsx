import React, { Component } from "react";
import { connect } from "react-redux";
import ExpenseForm from "./expenseForm";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";

export class EditPage extends Component {
  onRemove = () => {
    this.props.startRemoveExpense(this.props.expense);
    this.props.history.push("/");
  };

  onSubmit = expense => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };

  render() {
    return (
      <div>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
        <button className="btn btn-primary" onClick={this.onRemove}>
          Remove
        </button>
      </div>
    );
  }
}

// maps

const mapDispatchToProps = dispatch => {
  return {
    startEditExpense: (expenseID, expense) =>
      dispatch(startEditExpense(expenseID, expense)),
    startRemoveExpense: expense => dispatch(startRemoveExpense(expense))
  };
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(expense => {
      return expense.id === props.match.params.id;
    })
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditPage);
