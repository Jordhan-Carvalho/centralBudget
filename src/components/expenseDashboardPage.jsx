import React, { Component } from "react";
import { Link } from "react-router-dom";
import ExpenseList from "./expenseList";
import ExpenseListFilters from "./expenseListFilters";

class ExpenseDashboardPage extends Component {
  state = {};
  render() {
    return (
      <React.Fragment>
        <h1>Hello</h1>
        <ExpenseListFilters />
        <ExpenseList />
        <Link to="/create">Add expense</Link>
        <Link to="edit">edit expense</Link>
      </React.Fragment>
    );
  }
}

export default ExpenseDashboardPage;
