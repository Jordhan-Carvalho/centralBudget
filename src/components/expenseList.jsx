import React from "react";
import { connect } from "react-redux";
import ListItem from "./common/listItem";
import selectVisible from "../selectors/visibleTrans";

export const ExpenseList = props => {
  return (
    <div>
      <h1>Expense List</h1>
      {props.expenses.length === 0 ? (
        <p>No expenses</p>
      ) : (
        <ListItem type={"expense"} expenses={props.expenses} />
      )}
    </div>
  );
};

// redux call
const mapStateToProps = state => {
  return {
    expenses: selectVisible(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
