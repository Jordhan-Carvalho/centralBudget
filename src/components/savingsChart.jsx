import React, { Component } from "react";
import {
  ResponsiveContainer,
  ComposedChart,
  Area,
  Bar,
  Line,
  CartesianGrid,
  XAxis,
  YAxis,
  Tooltip,
  Legend
} from "recharts";
import { connect } from "react-redux";
import {
  getApril,
  getMay,
  getJune,
  getJuly,
  getAugust,
  getSept,
  getOctober,
  getNovember,
  getDezember
} from "../services/montlyChartData";

export class SavingsChart extends Component {
  state = {
    data: []
  };

  getExpenses = (exp, inc, sav) => {
    const data = [
      ...this.state.data,
      getApril(exp, inc, sav),
      getMay(exp, inc, sav),
      getJune(exp, inc, sav),
      getJuly(exp, inc, sav),
      getAugust(exp, inc, sav),
      getSept(exp, inc, sav),
      getOctober(exp, inc, sav),
      getNovember(exp, inc, sav),
      getDezember(exp, inc, sav)
    ];
    this.setState({ data });
  };

  async componentDidMount() {
    await this.getExpenses(
      this.props.expenses,
      this.props.incomes,
      this.props.savings
    );
  }

  componentDidUpdate(prevProps) {
    if (
      this.props.expenses !== prevProps.expenses ||
      this.props.incomes !== prevProps.incomes ||
      this.props.savings !== prevProps.savings
    ) {
      this.getExpenses(
        this.props.expenses,
        this.props.incomes,
        this.props.savings
      );
    }
  }

  render() {
    return (
      <ResponsiveContainer width="100%" height={this.props.height}>
        <ComposedChart
          data={this.state.data}
          margin={{ top: 10, right: 30, left: 0, bottom: 0 }}
        >
          <XAxis dataKey="name" />
          <YAxis />
          <Tooltip />
          <Legend />
          <CartesianGrid stroke="#f5f5f5" />
          <Legend verticalAlign="bottom" height={36} />
          <Area
            type="monotone"
            dataKey="expenses"
            fill="#8884d8"
            stroke="#8884d8"
          />
          <Bar dataKey="savings" barSize={20} fill="#413ea0" />
          <Line type="monotone" dataKey="incomes" stroke="#ff7300" />
        </ComposedChart>
      </ResponsiveContainer>
    );
  }
}

//redux
const mapStateToProps = state => {
  return {
    expenses: state.expenses,
    incomes: state.incomes,
    savings: state.savings
  };
};

export default connect(mapStateToProps)(SavingsChart);
