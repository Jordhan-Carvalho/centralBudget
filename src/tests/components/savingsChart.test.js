import React from 'react';
import { shallow } from 'enzyme';
import { SavingsChart } from '../../components/savingsChart';

test('should render SavingsChart correctly', () => {
    const wrapper = shallow(<SavingsChart height={400} />);
    expect(wrapper).toMatchSnapshot();
});