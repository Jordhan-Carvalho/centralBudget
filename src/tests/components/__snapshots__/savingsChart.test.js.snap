// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SavingsChart correctly 1`] = `
<ResponsiveContainer
  debounce={0}
  height={400}
  width="100%"
>
  <ComposedChart
    barCategoryGap="10%"
    barGap={4}
    data={Array []}
    layout="horizontal"
    margin={
      Object {
        "bottom": 0,
        "left": 0,
        "right": 30,
        "top": 10,
      }
    }
    reverseStackOrder={false}
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      dataKey="name"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={false}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      type="number"
      width={60}
      yAxisId={0}
    />
    <Tooltip
      active={false}
      animationDuration={400}
      animationEasing="ease"
      contentStyle={Object {}}
      coordinate={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      cursor={true}
      cursorStyle={Object {}}
      filterNull={true}
      isAnimationActive={true}
      itemSorter={[Function]}
      itemStyle={Object {}}
      labelStyle={Object {}}
      offset={10}
      separator=" : "
      useTranslate3d={false}
      viewBox={
        Object {
          "x1": 0,
          "x2": 0,
          "y1": 0,
          "y2": 0,
        }
      }
      wrapperStyle={Object {}}
    />
    <Legend
      align="center"
      iconSize={14}
      layout="horizontal"
      verticalAlign="bottom"
    />
    <CartesianGrid
      fill="none"
      horizontal={true}
      horizontalFill={Array []}
      horizontalPoints={Array []}
      stroke="#f5f5f5"
      vertical={true}
      verticalFill={Array []}
      verticalPoints={Array []}
    />
    <Legend
      align="center"
      height={36}
      iconSize={14}
      layout="horizontal"
      verticalAlign="bottom"
    />
    <Area
      activeDot={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey="expenses"
      dot={false}
      fill="#8884d8"
      fillOpacity={0.6}
      hide={false}
      isAnimationActive={true}
      legendType="line"
      points={Array []}
      stroke="#8884d8"
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
    <Bar
      animationBegin={0}
      animationDuration={400}
      animationEasing="ease"
      barSize={20}
      data={Array []}
      dataKey="savings"
      fill="#413ea0"
      hide={false}
      isAnimationActive={true}
      layout="vertical"
      legendType="rect"
      minPointSize={0}
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      xAxisId={0}
      yAxisId={0}
    />
    <Line
      activeDot={true}
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey="incomes"
      dot={true}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      points={Array []}
      stroke="#ff7300"
      strokeWidth={1}
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
  </ComposedChart>
</ResponsiveContainer>
`;
